runtime/src/qio/qio_formatted.c

```diff
@@ -360,6 +360,68 @@ qioerr _append_char(char* restrict * restrict buf, size_t* restrict buf_len, siz
 
   return 0;
 }
+qioerr read_entire_file(qio_channel_t* restrict ch, const char* restrict* restrict out, int64_t* restrict len_out, ssize_t maxlen)
+{
+    qioerr err,err_reading_file;
+    int64_t file_length;
+    err=qio_file_length(ch->file,&file_length); //extracting file size information to allocate buffer with approriate size beforehand
+    char* ret=NULL;
+    ssize_t amt;
+    err_t errcode;
+    if(!err)
+    {
+        
+        //allocate the buffer
+        ret=qio_malloc(file_length+1);
+        ret[0]='\0';
+        err_reading_file=qio_channel_read(false, ch, ret, file_length, &amt);
+        
+    }
+    else           //in case of socket or pipe files, file size info is not readily available.This has also been handled here .
+    {
+        
+        int64_t fileLenGuess=1024;
+        ret=qio_malloc(fileLenGuess+1);
+        while(true)
+        {
+            ret[0]='\0';
+            err_reading_file = qio_channel_read(false, ch, ret, fileLenGuess, &amt);
+            if (err_reading_file == QIO_EEOF) break;
+            fileLenGuess *= 2;      
+            ret = qio_realloc(ret, fileLenGuess);  //buffer reallocation since file contains more data than the buffer can hold .
+        }
+        file_length=fileLenGuess;
+    }
+    ret[file_length] = '\0'; // always add terminator at the end
+    if( err_reading_file ) goto rewind;
+    if( amt != file_length ) {
+        err_reading_file = QIO_ESHORT;
+        // zero out the rest of it...
+        memset(ret + amt, 0, file_length - amt);
+        goto rewind;
+      }
+
+          
+
+    err_reading_file = 0;
+    rewind:
```

Comment:
this rewind block should not be needed here because this function should be run before mark is called in the calling function

Reply:
Changes made as requested ,

Reply:
@mppf  Sir, request you to kindly have a look at this link - https://github.com/Sayan-Chaudhuri/chapel/pull/1/commits/952cf98a6c9e9c8fef27a8f501cfea140eab98c0.  I had already made the corrections and linked below with the name "comments added for requested changes"  .

Reply:
Yes, I see no rewind block when I look at that commit, but https://github.com/Sayan-Chaudhuri/chapel/pull/1/files still shows it. Can you try to figure out why?